server/src/main/java/org/elasticsearch/search/aggregations/bucket/BucketsAggregator.java

```diff
@@ -107,15 +108,33 @@ public final void collectExistingBucket(LeafBucketCollector subCollector, int do
      * Refer to that method for documentation about the merge map.
      */
     public final void mergeBuckets(long[] mergeMap, long newNumBuckets) {
```

Comment:
Could you deprecate this method? I *think* all callers would be better off with the other method.

Reply:
Done! I just marked it as deprecated. Should I also update all other usages of this method?

Reply:
Marking it as deprecated is plenty for this or!